pkg/ingester/ingester.go

```diff
@@ -1567,8 +1566,9 @@ func (i *Ingester) LabelNamesAndValues(request *client.LabelNamesAndValuesReques
 // We arbitrarily set it to 1mb to avoid reaching the actual gRPC default limit (4mb).
 const labelValuesCardinalityTargetSizeBytes = 1 * 1024 * 1024
 
-func (i *Ingester) LabelValuesCardinality(req *client.LabelValuesCardinalityRequest, srv client.Ingester_LabelValuesCardinalityServer) error {
-	if err := i.checkRunning(); err != nil {
+func (i *Ingester) LabelValuesCardinality(req *client.LabelValuesCardinalityRequest, srv client.Ingester_LabelValuesCardinalityServer) (lvcErr error) {
+	defer func() { lvcErr = i.handleReadError(lvcErr) }()
```

Comment:
Same comment as before, i.e. I think that the returned error should be named `err` to avoid confusion and potential programming mistakes.